File: backend/src/menus/entities/menu.entity.ts
import { MealTime } from '../dto/create-menu.dto';

export class Menu {
  id: number;
  subcategory_id: number;
  protein_type_id?: number;
  image_url?: string;
  contains: any; // JSON object
  meal_time: MealTime;
  is_active: boolean;
  created_at: Date;
  updated_at: Date;
}

export class MenuWithTranslations extends Menu {
  subcategory?: {
    id: number;
    name: string;
    category?: {
      id: number;
      name: string;
      food_type?: {
        id: number;
        name: string;
      };
    };
  };
  protein_type?: {
    id: number;
    name: string;
  };
  average_rating?: number;
  total_ratings?: number;
  name?: string;
}

File: frontend/lib/features/random_menu/domain/entities/menu_entity.dart
import 'package:equatable/equatable.dart';

/// Menu Entity - Pure business object
class MenuEntity extends Equatable {
  final int id;
  final String key;
  final String? imageUrl;
  final String mealTime;
  final bool isActive;
  final String name;
  final String? description;
  final Map<String, dynamic>? contains;
  final MenuSubcategoryEntity? subcategory;
  final MenuProteinTypeEntity? proteinType;
  final double? averageRating;
  final int? totalRatings;

  const MenuEntity({
    required this.id,
    required this.key,
    this.imageUrl,
    required this.mealTime,
    required this.isActive,
    required this.name,
    this.description,
    this.contains,
    this.subcategory,
    this.proteinType,
    this.averageRating,
    this.totalRatings,
  });

  @override
  List<Object?> get props => [
        id,
        key,
        imageUrl,
        mealTime,
        isActive,
        name,
        description,
        contains,
        subcategory,
        proteinType,
        averageRating,
        totalRatings,
      ];
}

class MenuSubcategoryEntity extends Equatable {
  final int id;
  final String key;
  final String name;
  final MenuCategoryEntity? category;

  const MenuSubcategoryEntity({
    required this.id,
    required this.key,
    required this.name,
    this.category,
  });

  @override
  List<Object?> get props => [id, key, name, category];
}

class MenuCategoryEntity extends Equatable {
  final int id;
  final String key;
  final String name;

  const MenuCategoryEntity({
    required this.id,
    required this.key,
    required this.name,
  });

  @override
  List<Object?> get props => [id, key, name];
}

class MenuProteinTypeEntity extends Equatable {
  final int id;
  final String key;
  final String name;

  const MenuProteinTypeEntity({
    required this.id,
    required this.key,
    required this.name,
  });

  @override
  List<Object?> get props => [id, key, name];
}
